app/src/main/java/com/android/shelfLife/ui/newoverview/NewHouseHoldCreationScreen.kt

```diff
@@ -0,0 +1,283 @@
+package com.android.shelfLife.ui.newoverview
+
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.outlined.Close
+import androidx.compose.material.icons.outlined.Delete
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.shelfLife.R
+import com.android.shelfLife.model.newCreationScreen.CreationScreenViewModel
+import com.android.shelfLife.model.newInvitations.InvitationRepository
+import com.android.shelfLife.model.newhousehold.HouseHoldRepository
+import com.android.shelfLife.model.user.UserRepository
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.ui.navigation.Screen
+import com.android.shelfLife.ui.newutils.DeletionConfirmationPopUp
+import com.android.shelfLife.ui.utils.CustomButtons
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun HouseHoldCreationScreen(
+    navigationActions: NavigationActions,
+    houseHoldRepository: HouseHoldRepository,
+    invitationRepository: InvitationRepository,
+    userRepository: UserRepository
+) {
+  val creationScreenViewModel: CreationScreenViewModel = viewModel {
+    CreationScreenViewModel(
+        houseHoldRepository = houseHoldRepository,
+        invitationRepository = invitationRepository,
+        userRepository = userRepository)
+  }
+  val householdToEdit by creationScreenViewModel.householdToEdit.collectAsState()
+
+  var isError by rememberSaveable { mutableStateOf(false) }
```

Comment:
wouldnt all these variables be reset if we change the phone mode between light and dark (the bug that we had talked about)?

Reply:
No, rememberSaveable keeps the values through configuration changes, I use them to keep smaller state objects that would bloat the viewModel